@xen-orchestra/fs/src/fs.spec.js

```diff
@@ -0,0 +1,143 @@
+/* eslint-env jest */
+
+import { getHandler } from '.'
+import { sep } from 'path'
+import fs from 'fs-extra'
+import os from 'os'
+
+const tmpDir = os.tmpdir()
+const handlerList = [
+  { url: 'file://' + tmpDir },
+  // {
+  //   url: 'smb://login:pass@WORKGROUP\\\\ip\\smb\u0000',
+  //   id: '12345',
+  // },
+  // {
+  //   url: 'nfs://ip:/tmp/test',
+  //   id: '12345',
+  // },
+]
+
+const testDir = `${tmpDir}${sep}tests_${Date.now()}`
+const pathFile = `${testDir}${sep}file`
+
+handlerList.forEach(element => {
+  let handler
+
+  beforeAll(async () => {
+    handler = getHandler(element)
+    await handler.outputFile(pathFile, '')
+  })
+
+  afterAll(async () => {
+    if (fs.existsSync(testDir)) {
+      await handler.rmdir(testDir, { recursive: true })
+    }
+    handler = null
```

Comment:
I hate the fact that JavaScript has both `null` and `undefined`, this leads to lot of confusions/errors.

I personally try to always use `undefined` unless I really need `null`.